src/main/java/persistence/entity/PersistenceContext.java

```diff
@@ -0,0 +1,13 @@
+package persistence.entity;
+
+import java.util.Optional;
+
+public interface PersistenceContext {
+
+    <T, ID> Optional<T> getEntity(ID id, Class<T> entityType);
+
+    <T> void addEntity(T entity);
+
+    <T> void removeEntity(T entity);
```

Comment:
제네릭 메소드가 활용되지 않는 로직은 Object를 사용하도록 변경해보는건 어떨까요? :)

```suggestion
    void addEntity(Object entity);

    void removeEntity(Object entity);
```